### ISSUE TITLE
About equation (9) in the paper.

### ISSUE THREAD

**Issue by YukunLi99** [linhaowei1/TPLR#5]
I find that the score is taken as the opposite number in the code. Why is the final score calculated in this way?
The equation (9) in the paper is:
![屏幕截图 2024-07-18 143313](https://github.com/user-attachments/assets/8ed4729e-3770-4f85-bf19-bc2493418a7a)
The following is the [code implementation](https://github.com/linhaowei1/TPL/blob/main/utils/baseline.py#L45):
```
composition = -torch.logsumexp(torch.stack((-e1, -e2), dim=0), dim=0)
```

**Comment by linhaowei1** [comment]
Hello,

This might be a legacy typo. The composition score simulating the OR operator uses the function $-logsumexp(-e_1-e_2)$ in [1], while we adopt a different definition of energy functions $p(x) = e^{E(x)}$ to align with the definition of the OOD score.

In our experiments, I recall that the results using `torch.logsumexp(torch.stack((e1, e2), dim=0), dim=0)` and `-torch.logsumexp(torch.stack((-e1, -e2), dim=0), dim=0)` were similar (though I am not completely certain as it has been some time since then). If you are interested, feel free to give it a try.

[1] Du et al. Compositional Visual Generation with Energy Based Models. NeurIPS 2020.
